crates/sui-proxy/src/metrics.rs

```diff
@@ -53,3 +99,20 @@ async fn metrics(Extension(registry_service): Extension<RegistryService>) -> (St
         ),
     }
 }
+
+/// pod_health is called by k8s to know if this service is correctly processing data
+async fn pod_health(Extension(pod_health): Extension<HealthCheckMetrics>) -> (StatusCode, String) {
```

Comment:
Nit: why not pass the `RegistryService` here? Then you wouldn't have to add the `HealthCheck` struct

Reply:
because i didn't want to trigger a metrics scrape.  when the metrics are scraped, the value i use here is updated and then acts as a cached value based on the last scrape interval.